packages/shared-ini-file-loader/src/slurpFile.ts

```diff
@@ -3,45 +3,19 @@ import { promises as fsPromises } from "fs";
 
 const { readFile } = fsPromises;
 
-type callbacks = { resolve: Function; reject: Function };
-type FileStatus = {
-  contents: string;
-  isReading: boolean;
-  requestQueue: callbacks[];
-};
-
-const fileStatusHash: { [key: string]: FileStatus } = {};
+const filePromisesHash: { [key: string]: Promise<string> } = {};
 
-export const slurpFile = (path: string) =>
-  new Promise<string>((resolve, reject) => {
-    if (!fileStatusHash[path]) {
-      // File not read yet, set file isReading to true and read file.
-      fileStatusHash[path] = { isReading: true, contents: "", requestQueue: [] };
-      fileStatusHash[path].requestQueue.push({ resolve, reject });
+export const slurpFile = (path: string) => {
+  if (!filePromisesHash[path]) {
+    filePromisesHash[path] = new Promise((resolve, reject) => {
```

Comment:
It looks like this `new Promise(...)` is now equivalent to
```js
filePromisesHash[path] = readFile(path, "utf8");
```

Reply:
Done in [`9f4170a` (#3546)](https://github.com/aws/aws-sdk-js-v3/pull/3546/commits/9f4170a861d117f75dc8b842eb4ed6d8d14f6c1f)